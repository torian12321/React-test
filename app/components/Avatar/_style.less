@import '~LESS_conf';


.avatar{
    background-size: cover;
// Zoom
    background: noRepeat center center fixed;
    //backgroundSize: 'cover';

    .rounded();
}