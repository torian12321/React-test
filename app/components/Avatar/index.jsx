var
React     = require('react'),
PropTypes = require('prop-types'),
styles    = require('./_style.less');

const Avatar = props => {
    let className = classNames(
        'avatar',
        props.size,
        props.className
    );

    return(
        <div 
            style    ={{ backgroundImage: `url(${props.src})`}}
            className={className}
            alt      ={props.alt}
            role     ={props.alt ? null : 'presentation'}
        />
    );
}


Avatar.propTypes = {
    src      : PropTypes.string.isRequired,
    className: PropTypes.string
};
Avatar.defaultProps = {
    src: 'http://lorempixel.com/200/200/people'
};

module.exports = Avatar;
