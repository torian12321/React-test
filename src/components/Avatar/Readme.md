#### Basic
```
<Avatar />
```

#### States
May be used for 'online', 'in a meeting' etc.
```
<div>
	<Avatar state='main'/>
    <Avatar state='sec'/>
    <Avatar state='success'/>
    <Avatar state='danger'/>
    <Avatar state='warning'/>
    <Avatar state='disable'/>
</div>
```

#### Sizes
```
<div>
	<Avatar size='xs'/>
	<Avatar size='sm'/>
	<Avatar size='md'/>
	<Avatar size='lg'/>
</div>
```