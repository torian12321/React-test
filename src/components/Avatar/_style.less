@import '~LESS_conf';

.avatar{
    @size : 80px;
    @color: @themeColor_2;

    font-size : @size;
    height    : 1em;
    width     : 1em;
    border    : @color 2px solid;
    background: @color center center no-repeat;
    box-sizing: border-box;
    .rounded(50%);
    .background-cover();

    // States
    &.main   {border-color: @themeColor_1;}
    &.sec    {border-color: @themeColor_2;}
    &.success{border-color: @colorState_success;}
    &.danger {border-color: @colorState_danger;}
    &.warning{border-color: @colorState_warning;}
    &.disable{border-color: @colorState_disable;}

    // Sizes
    &.xs{font-size: @size * 0.5;}
    &.sm{font-size: @size;}
    &.md{font-size: @size * 1.5;}
    &.lg{font-size: @size * 2.5;}
}