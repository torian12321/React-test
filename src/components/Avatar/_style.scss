@import '../../styles/conf/index.scss';

// Vars
$avatar-size : 80px !default;
$avatar-color: $themeColor_2 !default;

// Block
.avatar{
  font-size : $avatar-size;
  height    : 1em;
  width     : 1em;
  border    : $avatar-color 2px solid;
  background: $avatar-color;
  box-sizing: border-box;
  @include rounded(50%);
  @include background-cover();

  // Modifiers

  // States
  &--main {border-color: $themeColor_1;}
  &--sec  {border-color: $themeColor_2;}

  @each $stateName, $stateColor in $states {
    &--#{$stateName} {
      border-color: nth($stateColor, 1);
    }
  }

  // Sizes
  &--xs {font-size: $avatar-size * 0.5;}
  &--sm {font-size: $avatar-size;}
  &--md {font-size: $avatar-size * 1.5;}
  &--lg {font-size: $avatar-size * 2.5;}
}