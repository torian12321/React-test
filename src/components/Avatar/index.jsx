import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import ImgDef     from '../../img/default/user.png';
import './_style.scss';

class Avatar extends React.PureComponent {
  render() {
    return (
      <div 
        style    ={{backgroundImage: `url(${this.props.src})`}}
        className={classNames(
          'avatar',
          this.props.size,
          this.props.state,
          this.props.className
        )}
      />
    )
  }
}


Avatar.propTypes = {
  src      : PropTypes.string.isRequired,
  size     : PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
  state    : PropTypes.oneOf(['main', 'sec', 'success', 'danger', 'warning', 'disable']),
  className: PropTypes.string
};
Avatar.defaultProps = {
  src: ImgDef
};

module.exports = Avatar;
