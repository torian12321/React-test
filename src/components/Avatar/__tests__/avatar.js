import React    from 'react';
import renderer from 'react-test-renderer';
import Avatar   from '../index';

describe('component', () => {
  describe('<Avatar />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Avatar src='img.png' />
      )).toMatchSnapshot();
    });

    it('Should render small size', () => {
      expect(renderer.create(
        <Avatar src='img.png' size='sm' />
      )).toMatchSnapshot();
    });

    it('Should render small danger State', () => {
      expect(renderer.create(
        <Avatar src='img.png' state='danger' />
      )).toMatchSnapshot();
    });

  });
});
