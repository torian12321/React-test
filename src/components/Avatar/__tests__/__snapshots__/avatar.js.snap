// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Avatar /> Should render small danger State 1`] = `
<div
  className="avatar danger"
  style={
    Object {
      "backgroundImage": "url(img.png)",
    }
  }
/>
`;

exports[`component <Avatar /> Should render small size 1`] = `
<div
  className="avatar sm"
  style={
    Object {
      "backgroundImage": "url(img.png)",
    }
  }
/>
`;

exports[`component <Avatar /> should render correctly 1`] = `
<div
  className="avatar"
  style={
    Object {
      "backgroundImage": "url(img.png)",
    }
  }
/>
`;
